src/evict/evict_lru.c

```diff
@@ -423,6 +423,15 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
         return (0);
 
     if (!__wt_cache_stuck(session)) {
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
@y123456yz what we have to do here is weigh additional code complexity, in two files against the unlikely chance that the cache will un-stuck itself with only 20ms left on the timer. Now if a user configured a very short cache timeout I think this code would become more important, but generally cache timeouts are in the order of minutes not seconds.

Given we add a whole variable to the cache.h file I'd rather keep the code as is avoiding additional complexity. Instead of handling a case that only affects diagnostic code and is unlikely to trigger.